libs/wingc/src/parser.rs

```diff
@@ -637,6 +638,43 @@ impl<'s> Parser<'s> {
 		})
 	}
 
+	fn build_lift_statement(&self, statement_node: &Node, phase: Phase) -> DiagnosticResult<StmtKind> {
+		// Lift statements are only legal in inflight
+		if phase != Phase::Inflight {
+			return self.with_error("Lift statements are only allowed in inflight phase", statement_node);
+		}
```

Comment:
Is there a technical reason for this restriction? Could/Should this work?

```wing
// I could only think of a contrived example
let b = new cloud.Bucket();
lift {b: [get]} {
  let f = new cloud.Function(inflight () => {
    b.get("blah");
  })
}
```

Reply:
Interesting. Today the lift mechanism stores lift information per method (in your example the implicit `handle` method). So when specifying `lift` or when actually doing something like `preflight_bucket.put()` the lift visitor knows in which method we're in and stores the relevant information for that method.

What you're suggesting implies that you can specify explicit lifts that are assigned to multiple methods:
```wing
lift {bucket: get} {
  class Foo {
    inflight foo_m() {
      let b = bucket;
      b.get("hey");
    }
  }
  class Boo {
    inflight boo_m() {
      let b = bucket;
      b.get("hey");
    }
  }
```

I'll need to modify the lift visitor to get that working..
I suggest a future improvement, if at all.